recipes/mesonlsp/meta.yaml

```
@@ -0,0 +1,55 @@
+{% set name = "mesonlsp" %}
+{% set version = "4.2.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/JCWasmx86/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 4f67325d48ab26ac1c53bf95d5e1e0e9e48091e4a5aaa47cbb4c3b96c1a2cb4d
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ stdlib('c') }}
+    - meson
+    - ninja
+    - pkg-config
+    - sed  # [unix]
+  host:
+    - libarchive
+    - libcurl
+    - fmt  # [osx]
+    - lsprotocol
+    - nlohmann_json
+    - pkgconf
+    - pygls
+    - libuuid
+  run:
+    - fmt  # [osx]
+    - libarchive
+    - libcurl
+    - nlohmann_json
+    - pkgconf
+    - libuuid
```

Most of of these should come in via `run_exports` or are header-only.
```suggestion
  run:
    - pkgconf
```

---

Upon further examination of my local installation of `mesonlsp`, it is dynamically linking to `libpkgconf`.  So I added a `run_export` to `pkgconf` as well.  I believe this is okay to do here because if the user is using `pkgconf` only to find dependencies, they should put it in `build`, which as far as I know does not propagate `run_exports`.  On the other hand if they want to link to `libpkgconf`, they would need to put `pkgconf` in `host`, which would propagate `run_exports`.  If my assumption here is wrong I will revert that change!